src/main/daml/Daml/Finance/Instrument/Swap/Fpml/Util.daml

```diff
@@ -329,7 +327,13 @@ calculateFixPaymentClaimsFromSwapStream fixedRateSchedule s periodicSchedule cal
         mconcat $ zipWith createClaim (zip calculationSchedule notionals) paymentScheduleAligned
       claims = if issuerPaysLeg then claimAmounts else give claimAmounts
       claimsTagged = prepareAndTagClaims dateToDateClockTime [claims] "Fix rate payment"
-      allTaggedClaims = [claimsTagged]
+      allTaggedClaims = case s.principalExchanges of
+        None -> [claimsTagged]
+        Some principalExchanges -> [claimsTagged, principalClaimsTagged]
+          where
+            rateFixingDates = map (.adjustedStartDate) paymentScheduleAligned
```

Comment:
For my understanding, could you remind me what is an aligned payment schedule?

Reply:
It is a payment schedule where the periods have been aligned with the periods in the calculation schedule (of equal length, and the rate calculated in calcPeriod_i is always paid in paymentPeriod_i).